doc/changelog/02-specification-language/18915-master+fixes11902-restrict-univ-obligations.rst

```diff
@@ -0,0 +1,6 @@
+- **Fixed:**
+  Restriction of universes in universe polymorphic `Program`'s obligations
```

Comment:
"restriction" is internal jargon, for users we say something about not having unused universe variables

Reply:
"Universe polymorphic `Program`'s obligations are now generalized only over the universe variables that effectively occur in the obligation"?

Reply:
seems ok